### DISCUSSION TITLE
Querying a date property with NSPredicate (DotNet)

### DISCUSSION THREAD

**Discussion by jrlaff** [realm/realm-dotnet#2724] [Q&A] [answered]
Hi,

I have a Realm class with a list of objects:

```
public class Owner : RealmObject
{
    [...]
    [Backlink(nameof(...))]
    public IQueryable<Dog> Dogs { get; }
}
```

And the other Realm class with a `DateTimeOffset` property:

```
public class Dog : RealmObject
{
    [...]
    public DateTimeOffset SomeDateTime { get; set; } = DateTimeOffset.MinValue;
}
```

How can I perform an **NSPredicate**-based query that filters the results based on a comparison between `SomeDateTime` and `now`, in the IQueryable objects?

I am looking for something such as:

```
var results = myOwner.Dogs.Filter($"SomeDateTime <= -NOW-");
```
...Where **-NOW-** would be... well, right now :)

I can't find any documentation that describes what can be done with NSPredicates in Realm, except the very short cheatsheet (https://academy.realm.io/posts/nspredicate-cheatsheet/). Is there any?

Thanks.

**Comment by nirinchev** [comment, marked as answer]
You should be able to use `myOwner.Dogs.Filter("SomeDateTime <= $0", DateTimeOffset.UtcNow`.

**Reply by jrlaff** [reply]
Thank you. Anyone has a documentation to point to?

**Reply by nirinchev** [reply]
There are some examples [here](https://docs.mongodb.com/realm-sdks/dotnet/latest/reference/Realms.CollectionExtensions.html#Realms_CollectionExtensions_Filter__1_System_Linq_IQueryable___0__System_String_Realms_RealmValue___), but we should definitely updates our regular docs with more information about the string-based query syntax.
